Stratis.Bitcoin/Utilities/PeriodicTask.cs

```diff
@@ -4,50 +4,75 @@
 
 namespace Stratis.Bitcoin.Utilities
 {
+    /// <summary>Periodic task executes an application defined task in a loop with a specified delay between two runs.</summary>
     public interface IPeriodicTask
     {
+        /// <summary>Name of the application task. It is used for logging.</summary>
         string Name { get; }
 
+        /// <summary>
+        /// Executes the application task once.
+        /// </summary>
         void RunOnce();
+
+        /// <summary>
+        /// Configures and starts the loop in which the application task is periodically executed.
+        /// </summary>
+        /// <param name="cancellation">Cancellation token that allows the caller to stop the loop.</param>
+        /// <param name="refreshRate">Delay between two executions of the task.</param>
+        /// <param name="delayStart">If true, the first execution of the task is made after the <paramref name="refreshRate"/> time.</param>
+        /// <returns>This task to enable fluent code.</returns>
         PeriodicTask Start(CancellationToken cancellation, TimeSpan refreshRate, bool delayStart = false);
     }
 
+    /// <inheritdoc />
+    /// <remarks>This implementation spawns a new thread for each new periodic task.</remarks>
     public class PeriodicTask : IPeriodicTask
```

Comment:
The plan is to get rid of this class and only use the AsyncLoop, no point in holding a task that sleeps 90% of the time.